firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -232,22 +243,35 @@ private boolean shouldIgnoreCache(String event) {
                       .defaultIfEmpty(CampaignImpressionList.getDefaultInstance())
                       .onErrorResumeNext(Maybe.just(CampaignImpressionList.getDefaultInstance()));
 
+              Maybe<InstanceIdResult> getIID = taskToMaybe(firebaseInstanceId.getInstanceId());
+
               Function<CampaignImpressionList, Maybe<FetchEligibleCampaignsResponse>> serviceFetch =
-                  impressions ->
-                      taskToMaybe(apiClient.getFiams(impressions))
-                          .doOnSuccess(
-                              resp ->
-                                  Logging.logi(
-                                      String.format(
-                                          Locale.US,
-                                          "Successfully fetched %d messages from backend",
-                                          resp.getMessagesList().size())))
-                          .doOnSuccess(
-                              resp -> impressionStorageClient.clearImpressions(resp).subscribe())
-                          .doOnSuccess(analyticsEventsManager::updateContextualTriggers)
-                          .doOnSuccess(testDeviceHelper::processCampaignFetch)
-                          .doOnError(e -> Logging.logw("Service fetch error: " + e.getMessage()))
-                          .onErrorResumeNext(Maybe.empty()); // Absorb service failures
+                  campaignImpressionList -> {
+                    if (!dataCollectionHelper.isAutomaticDataCollectionEnabled()) {
+                      Logging.logi(
+                          "Automatic data collection is disabled, not attempting campaign fetch from service.");
+                      return Maybe.just(cacheExpiringResponse());
```

Comment:
Thinking out loud: Can this be modelled as an empty as well, similar to IID?

Reply:
hmm maybe but I think this is pretty clear

Reply:
like it could map to empty or something like that but I think it's not worth it